src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
+            **polyline** argument instead).
+        polyline: ``shapely`` polyline or collection of polylines (or `None` if using
+            the **points** argument instead).
+        orientation (optional): :term:`preferred orientation` to use, or `True` to use an
+            orientation aligned with the direction of the polyline (the default).
+        name (str; optional): name for debugging.
+    """
+    def __init__(self, points=None, polyline=None, orientation=True, name=None):
+        if orientation is True:
+            orientation = VectorField('Polyline', self.defaultOrientation)
+            self.usingDefaultOrientation = True
+        else:
+            self.usingDefaultOrientation = False
+        super().__init__(name, orientation=orientation)
+
+        if points is not None:
+            points = tuple(toVector(pt).coordinates for pt in points)
+
+            if any(point[2] != 0 for point in points):
+                warnings.warn('"points" passed to PolylineRegion have nonzero Z'
+                              ' components. These will be replaced with 0.')
+
+            points = tuple((x,y,0) for x,y,_ in points)
+
+            if len(points) < 2:
+                raise ValueError('tried to create PolylineRegion with < 2 points')
+            self.points = points
+            self.lineString = shapely.geometry.LineString(points)
+        elif polyline is not None:
+            if isinstance(polyline, shapely.geometry.LineString):
+                if len(polyline.coords) < 2:
+                    raise ValueError('tried to create PolylineRegion with <2-point LineString')
+            elif isinstance(polyline, shapely.geometry.MultiLineString):
+                if len(polyline.geoms) == 0:
+                    raise ValueError('tried to create PolylineRegion from empty MultiLineString')
+                for line in polyline.geoms:
+                    assert len(line.coords) >= 2
+            else:
+                raise ValueError('tried to create PolylineRegion from non-LineString')
+            self.lineString = polyline
+            self.points = None
+        else:
+            raise ValueError('must specify points or polyline for PolylineRegion')
+
+        if not self.lineString.is_valid:
+            raise ValueError('tried to create PolylineRegion with '
+                             f'invalid LineString {self.lineString}')
+        self.segments = self.segmentsOf(self.lineString)
+        cumulativeLengths = []
+        total = 0
+        for p, q in self.segments:
+            dx, dy = p[0] - q[0], p[1] - q[1]
+            total += math.hypot(dx, dy)
+            cumulativeLengths.append(total)
+        self.cumulativeLengths = cumulativeLengths
+        if self.points is None:
+            pts = []
+            last = None
+            for p, q in self.segments:
+                if last is None or p[:2] != last[:2]:
+                    pts.append(toVector(p).coordinates)
```

Comment:
What's the point of restricting to the first two coordinates (since we set the third to 0 above), and of passing to a vector and back again?